### PR TITLE
chore: add information about connection

### PR DESCRIPTION
**Pull request by Nebulis**

### CODE REVIEW

**Review by wmclaxton — APPROVED** [review]
This additional information is very helpful.

File: decentralised_rendering.md [hunk 1]
```diff
@@ -10,7 +10,38 @@ Accepted
...
+
+## Definitions
+The following terms will be used across the document:
+
+- `Document Viewer`: Website that loads a `Decentralised Renderer` through an iframe and provide down the certificate for the display.
+- `Decentralised Renderer` or `Document renderer`: Website responsible of displaying one or more certificates. Usually they are not usable outside a iframe, but keep in mind it's a plain website.
+- `Layout`: A template used to display a specific certificate in a document renderer. A document renderer is able to rendere multiple kind of certificates. In other words, a document rednerer supports multiple `Layouts`
+- `View`: A `Layout` may be composed of multiples view. For instance in Opencerts, You may have a view to display the certificate, and a view to display the transcripts.
+- `Action`: A plain Javascript object used for the communication between the document viewer and the document renderer
+
```
Comment (cavacado) [inline decentralised_rendering.md:25]: ah yes, these definitions will definitely be helpful.
minor spelling mistake at line 22. 
`In other words, a document rednerer supports multiple `Layouts``

File: decentralised_rendering.md [hunk 2]
```diff
@@ -105,13 +136,13 @@ const action = {
...
 
-## From frame to host actions
+## From frame to host actions (document renderer to document viewer)
 
 The following list of actions are made for iframe to communicate to the host (and thus must be handled by application embedding the iframe):
 
 ### UPDATE_HEIGHT
 
-This action provides the full content height of the iframe so that the host can adapt the automatically the size of the embedded iframe.
+This action provides the full content height of the iframe so that the host can adapt the automatically the size of the embedded iframe. It's worht to note that we hope to get rid of ths action one day. It was created only because we had difficulties to size correctly the iframe from the document viewer. IF you know a way to get rid of this, please get in touch with us, we will be happy to improve our architecture.
```
Comment (cavacado) [inline decentralised_rendering.md:145]: worht -> worth

File: decentralised_rendering.md [hunk 3]
```diff
@@ -10,7 +10,38 @@ Accepted
...
+- The document send the list of views to the document viewer using the `UPDATE_TEMPLATES` action. Usually this step happens right after the `RENDER_DOCUMENT` action.
+- The document viewer send the `View` to display using the `SELECT_TEMPLATE` action. Usually, it's an action triggered by a user.
+
+You will find below, the list of actions
+
+## Connection
+`Penpal` is responsible for establishing the connection between the document viewer and the document renderer. The message sent by penpal to establish the connection is out-of-scope of this document. However to establish the connection correctly, few things must be mentionned:
+- The document viewer and the document renderer must both provide one method only, called `dispatch`. This method is the only used for the communication of any `Actions` (see below for a description of the different actions)
+- The document viewer may have to support multiplt version of penpal. Indeed, there were a breaking change between penpal v4 and penpal v5, making [the version incompatible](https://github.com/Aaronius/penpal/issues/52). You can also decide to support a specific version of penpal. It's up to the implementer to decide.
```
Comment (cavacado) [inline decentralised_rendering.md:41]: should we emphasise on the words `may have` with bold letters? brings abit of attention to this particular point.
multiplt -> multiple

File: decentralised_rendering.md (hunk 1 above)
Comment (Nebulis) [reply decentralised_rendering.md:25]: done

File: decentralised_rendering.md (hunk 2 above)
Comment (Nebulis) [reply decentralised_rendering.md:145]: ok

File: decentralised_rendering.md (hunk 3 above)
Comment (Nebulis) [reply decentralised_rendering.md:41]: done

**Review by cavacado — APPROVED** [review]

**Pull request merged**